### PR TITLE
fix openness inference for empty tag

### LINKED ISSUE #6873: Error without "unnecessary" wildcard
If I remove my "unnecessary wildcard" my code does not work anymore :p 
```roc
module [foo]

okStr : Str -> Result Str []*
okStr = \str ->
    Ok str

foo =
    Result.try (okStr "aa") \str ->
        Err (MyErr str)
```
This produces the warning:
```
$ roc check examples/helloWorld.roc 

── UNNECESSARY WILDCARD in examples/helloWorld.roc ─────────────────────────────

This type annotation has a wildcard type variable (*) that isn't
needed.

3│  okStr : Str -> Result Str []*
                                ^

Annotations for tag unions which are constants, or which are returned
from functions, work the same way with or without a * at the end. (The
* means something different when the tag union is an argument to a
function, though!)

You can safely remove this to make the code more concise without
changing what it means.

────────────────────────────────────────────────────────────────────────────────

0 errors and 1 warning found in 10 ms

```
If I remove the `*` in `okStr : Str -> Result Str []*` I get:
```
── TYPE MISMATCH in examples/helloWorld.roc ────────────────────────────────────

This 2nd argument to try has an unexpected type:

8│>      Result.try (okStr "aa") \str ->
9│>          Err (MyErr str)

The argument is an anonymous function of type:

    Str -> [
        Err [MyErr Str],
        Ok b,
    ]

But try needs its 2nd argument to be:

    Str -> Result b []

────────────────────────────────────────────────────────────────────────────────

1 error and 0 warnings found in 10 ms
```

It appears the `Result Str []` is not inferred as `Result Str []*`, which would be the expected behavior.

### PR DESCRIPTION
**Pull request by Anton-4**
Fixes #6873

### CODE REVIEW

File: crates/compiler/uitest/tests/exhaustive/extend_uninhabited_without_opening_union.txt [hunk 1]
```diff
@@ -3,5 +3,5 @@ app "test" provides [main] to "./platform"
 walkHelp : {} -> [Continue {}, Break []]
 
 main = when walkHelp {} is
-#           ^^^^^^^^^^^ [Break [], Continue {}]
+#           ^^^^^^^^^^^ [Break []w_a, Continue {}]
```
Comment (Anton-4) [inline crates/compiler/uitest/tests/exhaustive/extend_uninhabited_without_opening_union.txt:6]: Is this change ok @ayazhafiz?

Comment (ayazhafiz) [reply crates/compiler/uitest/tests/exhaustive/extend_uninhabited_without_opening_union.txt:6]: yes this looks fine

File: crates/compiler/can/src/annotation.rs [hunk 2]
```diff
@@ -980,7 +992,12 @@ fn can_annotation_help(
                     }
 
                     None => Type::EmptyTagUnion,
-                }
+                }*/
+
+                Type::TagUnion(
+                    Default::default(),
+                    TypeExtension::from_type(ext_type, is_implicit_openness),
+                )
```
Comment (rtfeldman) [inline crates/compiler/can/src/annotation.rs:1000]: @Anton-4 Before merging, can we clean up the diff so it's just this one and the one in the .txt file? e.g. get rid of the block comment and `dbg`s etc. 😄

Comment (Anton-4) [reply crates/compiler/can/src/annotation.rs:1000]: Absolutely, that's why I marked the PR as WIP

**Review by rtfeldman — APPROVED** [review]

**Pull request merged**